pkgs/development/python-modules/torch/mk-darwin-check.nix

```diff
@@ -0,0 +1,38 @@
+{
+  cudaPackages,
+  libraries,
+  pythonPackages,
+}:
+(cudaPackages.writeGpuTestPython.override { python3Packages = pythonPackages; })
```

Comment:
This doesn't need `writeGpuTestPython` then (it adds a `gpuCheck` attribute with `requiredSystemFeatures` set to access CUDA in the sandbox).

FWIW you make this a simple

```
runCommand "test-torch-mps" { nativeBuildInputs = [ (python.withPackages ...) ]; } ''
  python << \EOF
  import torch
  ...
  EOF
''
```

This maybe has the undesired side-effect that getting a script to run outside the sandbox would take an extra step, but also I don't think anyone would actually need that...

Reply:
What do you think about this version? I made it to use writeShellApplcation and simply put the python script to separate file because it doesn't need any generated things from Nix side